### ISSUE TITLE
Zed syntax error causes in confusing error message about file input

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#5046]
## tl;dr

A community user was confused by this error message:

```
$ zq -version
Version: v1.14.0-2-g4dc62369

$ zq -Z 'fork(=> file "./csv_1.csv" | yield email | collect(this), => file "./csv_2.csv" | yield this["Recommended Email"])'
zq: no such file: fork(=> file "./csv_1.csv" | yield email | collect(this), => file "./csv_2.csv" | yield this["Recommended Email"])
```

## Details

Repro is with Zed commit 4dc6236. The issue was initially reported in a [community Slack thread](https://brimdata.slack.com/archives/CTT24051C/p1708106640649629).

Many `zq` command lines are unambiguous when the first argument is the Zed program text and the remaining arguments are file inputs. The example above presents a challenge because there's just a single argument, so `zq` needs to use some kind of heuristic to determine how to treat that single argument. It could be a filename (e.g., a simple command line like `zq MyFile.zng` may often be used to view a file's contents as ZSON) or, as was clearly the user's intent here, it could be a Zed program that specifies the inputs from inside using the `file` operator. In this case there was a syntax error in the Zed program (the `,` before the `=>`) so the heuristic fell back to treating the single input as a filename, and the current error message about this understandably confused the user.

We discussed this one as a group and there was some thought that the heuristic could maybe stand to be revisited and/or a new flag could be introduced similar to `jq -n`. However, as a short term improvement there was some consensus that we might get by just improving the error message here to be explicit about what the heuristic attempted to make it more obvious to the user in a situation like this that it's actually a Zed syntax error. Most of the error messages from the Zed CLI tooling are fairly terse, but this situation is tricky enough that it may be one where some verbosity is justified.

**Comment by philrz** [comment]
Verified in Zed commit 38763f8.

Revisiting the original repro example, the error message is much improved.

```
$ zq -version
Version: v1.14.0-16-g38763f82

$ zq -Z 'fork(=> file "./csv_1.csv" | yield email | collect(this), => file "./csv_2.csv" | yield this["Recommended Email"])'
zq: could not invoke zq with a single argument because:
 - a file could not be found with the name "fork(=> file \"./csv_..."
 - the argument could not be compiled as a valid Zed query due to parse error (column 58):
   fork(=> file "./csv_1.csv" | yield email | collect(this), => file "./csv_2.csv" | yield this["Recommended Email"])
                                                         === ^ ===
```

Thanks @mattnibs!
